src/App.jsx

```diff
@@ -14,70 +15,62 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  // let selected = '';
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood ? `${selectedGood} is selected` : 'No goods selected'}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+        {selectedGood && (
+        <button
+          data-cy="ClearButton"
+          type="button"
+          className="delete ml-3"
+          onClick={() => {
+            setSelectedGood('');
+          }}
+        />
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => {
+            const isGoodSelected = selectedGood === good;
+            const chooseGood = () => (isGoodSelected ? undefined : good);
+            return (
+            <tr
+              data-cy="Good"
+              className={isGoodSelected
+                ? 'has-background-success-light' : ''}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={isGoodSelected
+                    ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  className={isGoodSelected
+                    ? 'button is-info' : 'button'}
+                  onClick={() => {
+                    setSelectedGood(chooseGood());
+                  }}
+                >
+                  { isGoodSelected ? '-' : '+'}
+                </button>
+              </td>
```

Comment:
if you change almost every prop depending on some condition
better to rewrite it:

```
isGoodSelected 
 ? <button></button>
 : <button></button>
```